lib/modules/design_pattern_details/widgets/rich_markdown_body.dart

```diff
@@ -0,0 +1,109 @@
+import 'dart:developer';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_markdown/flutter_markdown.dart';
+
+import '../../../helpers/dart_sytax_highlighter.dart';
+import '../../../helpers/url_launcher.dart';
+
+class RichMarkdownBody extends StatelessWidget {
+  const RichMarkdownBody({this.data, super.key});
+
+  final String? data;
+
+  String parseSyntaxLanguage(String data) {
+    String parsed = data;
+    final RegExp codeSign = RegExp('`{3} *');
+    final RegExp pattern = RegExp(r'`{3} *\w+');
+    for (final RegExpMatch match in pattern.allMatches(data)) {
+      final String? lang = match.group(0)?.split(codeSign)[1];
+      parsed = parsed.replaceFirst(match.group(0).toString(), '```$lang');
+      log('lang: $lang');
+    }
+    return parsed;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+
+    final textTheme = theme.textTheme;
+    final bodyMedium = textTheme.bodyMedium;
+    final fontSize = theme.textTheme.bodyMedium?.fontSize ?? 14.0;
+
+    return MarkdownBody(
+      selectable: true,
+      fitContent: false,
+      styleSheet: MarkdownStyleSheet(
+        h1: textTheme.headlineMedium,
+        h2: textTheme.titleLarge,
+        h3: textTheme.titleMedium,
+        h4: textTheme.titleMedium,
+        h5: textTheme.titleMedium,
+        h6: textTheme.titleLarge,
+        p: bodyMedium,
+        blockSpacing: 8,
+        blockquotePadding: const EdgeInsets.symmetric(
+          horizontal: 12,
+          vertical: 8,
+        ),
+        horizontalRuleDecoration: BoxDecoration(
+          border: Border(
+            top: BorderSide(
+              color: theme.colorScheme.primary.withOpacity(0.2),
+              width: 3,
+            ),
+          ),
+        ),
+        blockquoteDecoration: BoxDecoration(
+          color: theme.colorScheme.surface.withOpacity(0.2),
+          border: Border(
+            left: BorderSide(
+              color: theme.colorScheme.surface,
+              width: 3,
+            ),
+          ),
+        ),
+        codeblockDecoration: BoxDecoration(
+          color: theme.colorScheme.surface,
+          borderRadius: BorderRadius.circular(5),
+          border: Border.all(
+            color: theme.colorScheme.outline.withOpacity(0.2),
+          ),
+        ),
+        a: bodyMedium?.copyWith(
+          color: theme.colorScheme.primary,
+          decoration: TextDecoration.underline,
+        ),
+        blockquote: bodyMedium?.copyWith(
+          color: theme.colorScheme.onSurface.withOpacity(0.6),
+        ),
+        code: bodyMedium?.copyWith(
```

Comment:
@mkobuolys  From the configurations available, it's not possible mainly because `code` doesn't have decoration yet, only has text style available for customization.
I tend to use a different font for code and codeBlock, namely `FiraCode` for a better visual difference.
But i'm not sure you want that at the moment.

The closest i've gotten to the image you shared is on the screenshot below:

- `code` and `codeBlock` font is `FiraCode`
- `code` background is highlighted _(tried to apply the same on your project but, not happening, don't know why yet)_
- 🚨 An issue is that when you select the text you can't see that `code` was selected, because the selection is behind the code highlight

_screenshot from a project of mine:_
<img width="501" alt="Screenshot 2023-05-28 at 01 17 09" src="https://github.com/mkobuolys/flutter-design-patterns/assets/59662912/90c5b612-affc-4ceb-b12b-231b3e16ba0c">

Reply:
Posted info here -> https://github.com/mkobuolys/flutter-design-patterns/pull/52#issuecomment-1565771800

Reply:
Replied to your comment, check it in the PR.